vraptor-core/src/main/java/br/com/caelum/vraptor/ioc/cdi/extension/EagerApplicationScopedExtension.java

```diff
@@ -0,0 +1,38 @@
+package br.com.caelum.vraptor.ioc.cdi.extension;
+
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.util.AnnotationLiteral;
+
+import br.com.caelum.vraptor.ioc.cdi.EagerInitialized;
+
+/**
+ * When enabled, it allows users to use {@link EagerInitialized} annotation 
+ * to get your application scoped beans initialized just after CDI deployment.
+ *
+ * @author Rodrigo Turini
+ */
+public class EagerApplicationScopedExtension implements Extension {
+	
+	@SuppressWarnings("serial")
+	public void initialize(@Observes AfterDeploymentValidation event, BeanManager beanManager) {
+		
+		Set<Bean<?>> beans = beanManager.getBeans(Object.class, 
+				new AnnotationLiteral<EagerInitialized>() {});
+		
+		for (Bean<?> bean : beans) {
+			Class<?> beanClass = bean.getBeanClass();
+			if(beanClass.isAnnotationPresent(ApplicationScoped.class)){
+				CreationalContext<?> ctx = beanManager.createCreationalContext(bean);
+				beanManager.getReference(bean, beanClass, ctx).toString();
```

Comment:
Yes, we have to evolve :).

Em Wed Feb 04 2015 at 11:08:34 AM, Rodrigo Turini <notifications@github.com>
escreveu:

> In
> vraptor-core/src/main/java/br/com/caelum/vraptor/ioc/cdi/extension/EagerApplicationScopedExtension.java
> <https://github.com/caelum/vraptor4/pull/937#discussion_r24081320>:
>
> > + *
> > + * @author Rodrigo Turini
> > + */
> > +public class EagerApplicationScopedExtension implements Extension {
> > +	
> > +	@SuppressWarnings("serial")
> > +	public void initialize(@Observes AfterDeploymentValidation event, BeanManager beanManager) {
> > +		
> > +		Set<Bean<?>> beans = beanManager.getBeans(Object.class,
> > +				new AnnotationLiteral<EagerInitialized>() {});
> > +		
> > +		for (Bean<?> bean : beans) {
> > +			Class<?> beanClass = bean.getBeanClass();
> > +			if(beanClass.isAnnotationPresent(ApplicationScoped.class)){
> > +				CreationalContext<?> ctx = beanManager.createCreationalContext(bean);
> > +				beanManager.getReference(bean, beanClass, ctx).toString();
>
> I'd entirely agree with you if we didn't have so many servlet container
> users :)
> (including us, in our projects)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caelum/vraptor4/pull/937/files#r24081320>.
>